Enhancement - Added the ability to allow for Multi Sort Functionality in getContent

Raistlfiren opened the pull request:
Modified the legacy and upcoming storage system to allow for multi sort in the setcontent query. Now queries can be used as follows to sort on multiple columns. 

Ex.:
```twig
{% setcontent 'entity' orderby('-name, title') %}
```

**Edit**
Fixes #1255
Fixes #2234

GawainLynch commented on src/Storage/Query/Handler/OrderHandler.php:
```diff
@@ -17,12 +17,41 @@ class OrderHandler
+     */
+    protected function isMultiOrderQuery($order)
+    {
+        return ( strpos($order, ',') !== FALSE ? TRUE : FALSE );
```
Per [PSR-2](https://github.com/php-fig/fig-standards/blob/master/accepted%2FPSR-2-coding-style-guide.md#25-keywords-and-truefalsenull) the booleans should be lower case.

GawainLynch commented on src/Legacy/Storage.php:
```diff
@@ -1915,28 +1915,72 @@ private function getFieldName($name)
+     */
+    protected function isMultiOrderQuery($order)
+    {
+        return ( strpos($order, ',') !== FALSE ? TRUE : FALSE );
```
Per [PSR-2](https://github.com/php-fig/fig-standards/blob/master/accepted%2FPSR-2-coding-style-guide.md#25-keywords-and-truefalsenull) the booleans should be lower case.

GawainLynch commented on src/Legacy/Storage.php:
```diff
@@ -1915,28 +1915,72 @@ private function getFieldName($name)
+            if (empty($name)) {
+                return false;
+            }
+            if (strpos($name, 'RAND') !== false) {
```
This won't work cross platform.

GawainLynch commented:
Ah, found some stuff I missed in review… I'll add some comments for you to adjust if you don't mind…

GawainLynch commented on src/Storage/Query/Handler/OrderHandler.php:
```diff
@@ -17,12 +17,41 @@ class OrderHandler
+        $separatedOrders[] = $order;
+
+        if ($this->isMultiOrderQuery($order))
+            $separatedOrders = explode(",", $order);
```
Per [PSR-2 Section 5](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#5-control-structures) can you update this to:
```php
        if ($this->isMultiOrderQuery($order)) {
            $separatedOrders = explode(',', $order);
        }
```

GawainLynch commented on src/Storage/Query/Handler/OrderHandler.php:
```diff
@@ -17,12 +17,41 @@ class OrderHandler
+     */
+    protected function getOrderBys($order)
+    {
+        $separatedOrders[] = $order;
```
This one makes Scrutinizer unhappy, you're better off with:
```php
        $separatedOrders = [$order];
```

GawainLynch commented on src/Legacy/Storage.php:
```diff
@@ -1915,28 +1915,72 @@ private function getFieldName($name)
+        $separatedOrders[] = $order;
+
+        if ($this->isMultiOrderQuery($order))
+            $separatedOrders = explode(",", $order);
```
Per [PSR-2 Section 5](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#5-control-structures) can you update this to:
```php
        if ($this->isMultiOrderQuery($order)) {
            $separatedOrders = explode(',', $order);
        }
```

GawainLynch commented on src/Legacy/Storage.php:
```diff
@@ -1915,28 +1915,72 @@ private function getFieldName($name)
+     */
+    protected function getOrderBys($order)
+    {
+        $separatedOrders[] = $order;
```
Same as below:
```php
        $separatedOrders = [$order];
```

Raistlfiren commented:
This should also close this Enhancement issue - https://github.com/bolt/bolt/issues/2234

Raistlfiren commented:
I debated on putting this in another PR, but I realized that the function I modified relied on some of the methods I added in this PR. Anyway, I added the ability to also sort on multiple columns in the content type. That should close the enhancement issue - https://github.com/bolt/bolt/issues/1255

GawainLynch commented:
Awesome, thanks @Raistlfiren :+1:

GawainLynch commented:
Test added in https://github.com/bolt/bolt/commit/7da6b1cd361ef78c54c7e301049bb68b1e190a18
